study.md

```diff
@@ -21,18 +21,17 @@ There is no requirement to follow any links from these pages.
 After doing all the required readings please tell us anything you were confused about.
 
 ```md
-<!-- your answer here -->
+<!--Just want to clarify on callback functions, does it just mean a nested function? Like if, in math terms, I wanted to solve f(g(x)) is g(x) the callback function? And if it is, am I allowed to call g(x) by itself at any point after that or can I always call g(x) back inside another function?-->
 ```
 
 Are there any nagging questions relating to functions that you have?
-
 ```md
-<!-- your answer here -->
+<!--Can you go over anonymous functions? I think not getting that is part of why I'm having a hard time getting callback functions... like how do we call it back when it doesn't have a name?-->
```

Comment:
An anonymous function is one that has no name attached to it.  For example, arrow functions and unnamed functions that are used in callbacks are anonymous functions.  Here's an example.

```javascript
const array = ['a', 'b', 'c'];

// Anonymous function as a function declaration.
array.forEach(function(element, index) {
  console.log(`The element of the array at index ${index} is ${element}.`);
});
// Output:
// The element of the array at index 0 is a.
// The element of the array at index 1 is b.
// The element of the array at index 2 is c.

// Anonymous function as a arrow function.
array.forEach((element, index) => {
  console.log(`The element of the array at index ${index} is ${element}.`);
});
// Output:
// The element of the array at index 0 is a.
// The element of the array at index 1 is b.
// The element of the array at index 2 is c.

// Function declaration for showElementInfo().
function showElementInfo(element, index) {
  console.log(`The element of the array at index ${index} is ${element}.`);
}

// Named function as a callback function.
array.forEach(showElementInfo);
// Output:
// The element of the array at index 0 is a.
// The element of the array at index 1 is b.
// The element of the array at index 2 is c.

// As you can see, the outputs for all three of these are identical.
```